packages/now-cli/src/util/dev/server.ts

```diff
@@ -826,6 +832,13 @@ export default class DevServer {
     // Wait for "ready" event of the watcher
     await once(this.watcher, 'ready');
 
+    // Configure the server to forward WebSocket "upgrade" events to the proxy.
+    this.server.on('upgrade', (req, socket, head) => {
+      this.proxy.ws(req, socket, head, {
+        target: `http://localhost:${this.devProcessPort}`,
+      });
+    });
+
```

Comment:
Wondering how y'all feel about placement for this. `start` was my best guess, but I wasn't sure on that (it _could_ go in the constructor, I think?), and I wasn't sure this was the right place in `start`.

Setting up event handlers felt start-y to me, and I noticed not much other substantive things happening in the constructor… though I suppose we could refactor the callback here to be a method, and then configuring it in the constructor would be pretty concise?

Reply:
I think `start()` is fine. The only better place might be immediately after the port is assigned. But let's keep it here for now 👍